ext/mysqli/mysqli.c

```diff
@@ -1004,35 +1004,18 @@ PHP_FUNCTION(mysqli_stmt_construct)
 	MY_STMT				*stmt;
 	MYSQLI_RESOURCE		*mysqli_resource;
 	char				*statement;
-	size_t					statement_len;
+	size_t					statement_len = 0;
```

Comment:
Nullable might be better here? Unless maybe mysql_stmt_prepare already ignores empty strings.

Also wondering how this convoluted code came about...